dive_core/gfxr_vulkan_command_hierarchy.cpp

```diff
@@ -113,13 +120,29 @@ uint64_t GfxrVulkanCommandHierarchyCreator::AddNode(NodeType type, std::string &
 {
     uint64_t node_index = m_command_hierarchy.AddGfxrNode(type, std::move(desc));
 
-    for (uint32_t i = 0; i < CommandHierarchy::kTopologyTypeCount; ++i)
+    if (m_used_in_mixed_command_hierarchy)
     {
-        DIVE_ASSERT(m_node_children[i].size() == node_index);
-        m_node_children[i].resize(m_node_children[i].size() + 1);
+        uint64_t local_node_index = m_local_node_count++;
+        m_dive_indices_to_local_indices_map[node_index] = local_node_index;
 
-        m_node_root_node_index[i].resize(m_node_root_node_index[i].size() + 1);
+        for (uint32_t i = 0; i < CommandHierarchy::kTopologyTypeCount; ++i)
+        {
+            DIVE_ASSERT(m_node_children[i].size() == local_node_index);
```

Comment:
local_node_index == m_node_children[i].size(), do we still need the m_local_node_count to track the node count?

Reply:
Done. Also since the gfxr nodes are only added to the gpu events topology, there is no need for a loop.

Reply:
thanks